### PR TITLE
Implemented Linear Kernel algorithm for new interfaces

### PR DESCRIPTION
**Pull request by PetrovKP**
1. Added common interfaces of compute for all algorithms
2. Added Interface for compute for Linear Kernel
3. Created example for Linear Kernel

❓ Naming parameters `x`, `y`, `k` and `b` can changed
⚡ Support for GPU and DPC++ will be next PR

### CODE REVIEW

**Comment by PetrovKP** [comment]
@rlnx , @michael-smirnov , @makart19 , @mchernov-intel , @amgrigoriev Please, look this PR

File: cpp/oneapi/dal/algo/linear_kernel/backend/cpu/compute_kernel_default_dense.cpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+namespace interop            = dal::backend::interop;
+
+template <typename Float, daal::CpuType Cpu>
+using daal_linear_kernel_t =
+    daal_linear_kernel::internal::KernelImplLinear<daal_linear_kernel::defaultDense, Float, Cpu>;
+
+template <typename Float>
+static compute_result call_daal_kernel(const context_cpu& ctx,
+                                       const descriptor_base& desc,
+                                       const table& x,
```
Comment (mchernov-intel) [inline cpp/oneapi/dal/algo/linear_kernel/backend/cpu/compute_kernel_default_dense.cpp:37]: `x` and `y` are not suitable names in my opinion.

File: cpp/oneapi/dal/algo/linear_kernel/common.cpp [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/algo/linear_kernel/common.hpp"
+
+namespace oneapi::dal::linear_kernel {
+
+class detail::descriptor_impl : public base {
+public:
+    double k = 1.0;
```
Comment (mchernov-intel) [inline cpp/oneapi/dal/algo/linear_kernel/common.cpp:23]: Please, replace the names or provide a proof that it's suitable name.

File: cpp/oneapi/dal/algo/linear_kernel/common.cpp [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+double descriptor_base::get_k() const {
+    return impl_->k;
+}
+
+double descriptor_base::get_b() const {
+    return impl_->b;
+}
+
+void descriptor_base::set_k_impl(const double value) {
```
Comment (mchernov-intel) [inline cpp/oneapi/dal/algo/linear_kernel/common.cpp:39]: `const` doesn't matter. delete this everywhere, or force everyone to add `const` to object's sent by value.

File: cpp/oneapi/dal/data/array.hpp [hunk 4]
```diff
@@ -17,6 +17,7 @@
 #pragma once
 
 #include <algorithm>
+#include <stdexcept>
```
Comment (mchernov-intel) [inline cpp/oneapi/dal/data/array.hpp:20]: WTF?!

**Comment by mchernov-intel** [comment]
@PetrovKP please, get @rlnx approval regarding `DAAL_EXPORT`

**Comment by PetrovKP** [comment]
Move changes to https://github.com/oneapi-src/oneDAL/pull/695

**Comment by PetrovKP** [comment]
Move changes to #695

**Pull request closed without merging**